### PR TITLE
[stable-2.8] Fix sanity tests based on newer version of shellcheck (#60423)

### PR DESCRIPTION
**Pull request by samdoran**
##### SUMMARY

Backport of #60423 for Ansible 2.8

(cherry picked from commit c485a1b91e595234acdd2b57a5abe883c6306a35)

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

`test/integration/targets/blocks/runme.sh`
`test/integration/targets/group/files/grouplist.sh`
`test/integration/targets/handlers/runme.sh`
`test/integration/targets/include_import/runme.sh`
`test/integration/targets/inventory_vmware_vm_inventory/runme.sh`
`test/integration/targets/pull/runme.sh`
`test/integration/targets/user/files/userlist.sh`
`test/lib/ansible_test/_data/sanity/shellcheck/exclude.txt`

### CODE REVIEW

**Review by mattclay — APPROVED** [review]

**Comment by abadger** [comment]
I don't see inventory_cache as a shell variable in 2.8's version of test/integration/targets/inventory_vmware_vm_inventory/runme.sh  I'm going to rerun the tests and see if that fixes things.

File: test/integration/targets/inventory_vmware_vm_inventory/runme.sh [hunk 1]
```diff
@@ -49,22 +49,33 @@ curl "http://${VCENTER_HOST}:5000/govc_find"
 # Get inventory
 ansible-inventory -i ${VMWARE_CONFIG} --list
 
+echo "Check if cache is working for inventory plugin"
+if [ ! -n "$(find "${inventory_cache}" -maxdepth 1 -name 'vmware_vm_inventory_*' -print -quit)" ]; then
```
Comment (abadger) [inline test/integration/targets/inventory_vmware_vm_inventory/runme.sh]: In this old version of inventory_vmware_vm_inventory/runme.sh, ${inventory_cache } isn't defined.  It was defined in 8ce09a0057189733c814b01d0ecfa7ec405014a0  and it looks like you can just use ```$(pwd)/inventory_cache``` if you don't want to backport that.

**Comment by abadger** [comment]
Oh, I see.... $inventory_cache is being introduced in this PR.  So this is a valid bug in this PR.  Added my research on that to a comment inline.

File: test/integration/targets/inventory_vmware_vm_inventory/runme.sh (hunk 1 above)
Comment (samdoran) [reply test/integration/targets/inventory_vmware_vm_inventory/runme.sh]: Thanks for tracking this down. I pushed a fix.

**Comment by abadger** [comment]
THis is a test-only change.  Merging for 2.8.4 release.

**Pull request merged**